src/routes/article.ts

```diff
@@ -0,0 +1,42 @@
+import clownface from 'clownface';
+import createHttpError from 'http-errors';
+import { OK } from 'http-status-codes';
+import { addAll } from 'rdf-dataset-ext';
+import {
+  DefaultStateExtends, Middleware, Next,
+} from 'koa';
+import url from 'url';
+import Articles from '../articles';
+import { namedNode } from '../data-factory';
+import ArticleNotFound from '../errors/article-not-found';
+import { DatasetContext } from '../middleware/dataset';
+
+export default (articles: Articles): Middleware<DefaultStateExtends, DatasetContext> => (
+  async ({ request, response }: DatasetContext, next: Next): Promise<void> => {
```

Comment:
(Assuming this doesn't become a router middleware) Use `AppState` and `AppServiceContext`, and so `articles` from the context.

Reply:
We have the issue of `Dependency cycle detected.`

Reply:
We would need `AppConext` to access `path` from the context.